### PR TITLE
Fix 5478 move Swift settings to domain.xml for better governability, no disk access of properties file, leaner class

### PR DESCRIPTION
**Pull request by carlosmcgregor**
## Related Issues

- connects to #5478: Refactor Swift storage driver properties / properties getter

## Pull Request Checklist

- [x] Unit [tests][] completed
- [x] Integration [tests][]: None
- [x] Deployment requirements, [SQL updates][], [Solr updates][], etc.: None
- [x] [Documentation][docs] completed
- [x] Merged latest from "develop" [branch][] and resolved conflicts

[tests]: http://guides.dataverse.org/en/latest/developers/testing.html
[SQL updates]: http://guides.dataverse.org/en/latest/developers/sql-upgrade-scripts.html
[Solr updates]: https://github.com/IQSS/dataverse/blob/develop/conf/solr/7.3.0/schema.xml
[docs]: http://guides.dataverse.org/en/latest/developers/documentation.html
[branch]: http://guides.dataverse.org/en/latest/developers/branching-strategy.html

## Changes to Documentation

The following commands are now needed to configure _Swift_:

```
asadmin create-jvm-options "\-Ddataverse.file.swift-default-endpoint=endpoint1"
asadmin create-jvm-options "\-Ddataverse.file.swift-auth-type.endpoint1=$AUTH_TYPE"
asadmin create-jvm-options "\-Ddataverse.file.swift-auth-url.endpoint1=$AUTH_URL"
asadmin create-jvm-options "\-Ddataverse.file.swift-tenant.endpoint1=$TENANT"
asadmin create-jvm-options "\-Ddataverse.file.swift-username.endpoint1=$USERNAME"
asadmin create-jvm-options "\-Ddataverse.file.swift-password.endpoint1='${ALIAS=$PASSWORD}'"
asadmin create-jvm-options "\-Ddataverse.file.swift-endpoint.endpoint1=$ENDPOINT"
asadmin create-jvm-options "\-Ddataverse.file.swift-hash-key.endpoint1=$HASH_KEY"
```

It makes use of the `create-password-alias` approach that @pdurbin pointed out (thank you!). Otherwise, all settings remain the same as with `swift.properties`. Some additional settings that were hardcoded (i.e. separator and temporary URL expiry time) were moved to the settings. These are:

```
dataverse.files.swift-folder-path-separator
dataverse.files.swift-temporary-url-expiry-time
```

Please let me know if you have further questions regarding the changes that were made!

### CODE REVIEW

**Comment by carlosmcgregor** [comment]
@pdurbin, I'm on it!

**Comment by poikilotherm** [comment]
I am not sure if it's a good idea to move this to domain XML. For future platforms like Payara or Glassfish 5, the rising usage of configuration frameworks and more, this might be a long-term decision which is not easy to revert.

At least pretty please avoid "-" in the config names. These are invalid chars for environment variables. See IQSS/dataverse-kubernetes#11 Best option is to use a "." for namespacing as used elsewhere.

File: doc/sphinx-guides/source/installation/config.rst [hunk 1]
```diff
@@ -208,16 +208,24 @@ First, run all the following create commands with your Swift endpoint informatio
...
 .. code-block:: none
 
-    ./asadmin $ASADMIN_OPTS create-jvm-options "\-Ddataverse.file.swift-default-endpoint=endpoint1"
-    ./asadmin $ASADMIN_OPTS create-jvm-options "\-Ddataverse.file.swift-auth-type.endpoint1=your-auth-type"
-    ./asadmin $ASADMIN_OPTS create-jvm-options "\-Ddataverse.file.swift-auth-url.endpoint1=your-auth-url"
-    ./asadmin $ASADMIN_OPTS create-jvm-options "\-Ddataverse.file.swift-tenant.endpoint1=your-tenant-name"
-    ./asadmin $ASADMIN_OPTS create-jvm-options "\-Ddataverse.file.swift-username.endpoint1=your-username"
-    ./asadmin $ASADMIN_OPTS create-jvm-options "\-Ddataverse.file.swift-password.endpoint1='${ALIAS=your-password}'"
-    ./asadmin $ASADMIN_OPTS create-jvm-options "\-Ddataverse.file.swift-endpoint.endpoint1=your-swift-endpoint"
+    ./asadmin $ASADMIN_OPTS create-jvm-options "\-Ddataverse.file.swift.defaultEndpoint=endpoint1"
```
Comment (poikilotherm) [inline doc/sphinx-guides/source/installation/config.rst]: Some typos here. In the docs you use `.file.`, but in the code `.files.` is used.

Comment (pdurbin) [reply doc/sphinx-guides/source/installation/config.rst]: @poikilotherm good catch

Comment (carlosmcgregor) [reply doc/sphinx-guides/source/installation/config.rst]: Thank you for noticing that! Fixed.

**Comment by carlosmcgregor** [comment]
I read http://irclog.iq.harvard.edu/dataverse/2019-03-25#i_88874 and I would like to add one more setting that has `-`:

```
dataverse.files.storage-driver-id
```

**Comment by carlosmcgregor** [comment]
Sorry! Closed by accident :/

**Comment by pdurbin** [comment]
> I would like to add one more setting

@carlosmcgregor no problem, I pulled this out of code review until you're done. Please let us know when we should take another look. Thanks! Also, you're welcome to join us in IRC if you'd like. 😄

**Comment by carlosmcgregor** [comment]
Sorry, @pdurbin ! I should have added that `dataverse.files.storage-driver-id` is out of the scope of this pull. Just mentioned it so that it can be added to a list of Dataverse JVM options that need to be changed :)

**Comment by pdurbin** [comment]
@carlosmcgregor oh! I understand now. It looks like @poikilotherm is handling some of those "dash" JVM options over at https://github.com/IQSS/dataverse-kubernetes/blob/655b92d60b1a09e109154c9c237525827dbf4dff/dataverse-k8s/bin/init_1_configure.sh#L94

I'll move this issue and pull request back to code review. Thanks for explaining the scope.

**Comment by poikilotherm** [comment]
@carlosmcgregor Feel free to leave a comment on IQSS/dataverse-kubernetes#11 :-)

**Comment by carlosmcgregor** [comment]
Thank you @pdurbin and @poikilotherm !

**Comment by kcondon** [comment]
@carlosmcgregor Hi, woudl you mind refreshing this branch from /develop? We've since released v4.12 and your branch shows 4.11. This is causing a build issue. Once this is adjusted I will test and merge. Thanks! Kevin

**Comment by carlosmcgregor** [comment]
Done, @kcondon ! Thank you :)

**Pull request merged**